src/v/http/tests/http_imposter.cc

```diff
@@ -43,9 +43,20 @@ http_imposter_fixture::get_requests() const {
     return _requests;
 }
 
+static ss::sstring remove_query_params(std::string_view url) {
+    auto q_pos = url.find('?');
+    return q_pos == std::string_view::npos ? ss::sstring{url.data(), url.size()}
+                                           : ss::sstring{url.substr(0, q_pos)};
```

Comment:
`substr` with `npos` as count returns the full string, so maybe a simpler (although arguably, less clear) way to write this could be:

```
return ss::sstring{url.substr(0, url.find('?'))};
```

Reply:
i find it simpler to understand, added it thanks